src/App.tsx

```diff
@@ -1,39 +1,76 @@
 import React from 'react';
 import './App.scss';
+import '@fontsource/roboto/300.css';
+import '@fontsource/roboto/400.css';
+import '@fontsource/roboto/500.css';
+import '@fontsource/roboto/700.css';
+import Typography from '@mui/material/Typography';
+import AccessTimeIcon from '@mui/icons-material/AccessTime';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+interface State {
+  hasClock: boolean,
+  clockName: string,
+}
+
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  timeInterval: number | undefined;
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentDidMount(): void {
+    document.addEventListener('click', this.leftClickHandler);
+    document.addEventListener('contextmenu', this.rightClickHandler);
+    const interval = window.setInterval(() => this.setState(
+      { clockName: getRandomName() },
+    ), 3300);
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    this.timeInterval = interval;
+  }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  componentWillUnmount(): void {
+    document.removeEventListener('click', this.leftClickHandler);
+    document.removeEventListener('contextmenu', this.rightClickHandler);
+    window.clearInterval(this.timeInterval);
+  }
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+  leftClickHandler = () => {
```

Comment:
functions should always start with a verb
```suggestion
  handleRightClick = () => {
```